### PR TITLE
[SYCL] Don't synchronize last event if it is discarded

### PR DESCRIPTION
**Pull request by againull**

### CODE REVIEW

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -5921,7 +5921,7 @@ pi_result _pi_queue::synchronize() {
     return PI_SUCCESS;
 
   // For in-order queue just wait for the last command.
-  if (isInOrderQueue()) {
+  if (isInOrderQueue() && !LastCommandEvent->IsDiscarded) {
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:5924]: Why? Isn't `ZeEvent` still there and can be synced?

Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:5924]: Oh, I guess this is because it may not be host-scope right?

Comment (againull) [reply sycl/plugins/level_zero/pi_level_zero.cpp:5924]: Yes, this is one of the reasons. And the main reason is that this event may be in reset state. Basically we can't explicitly synchronize discarded pi_event objects.

Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:5924]: OK, but please add some comment about it.

**Review by smaslov-intel — APPROVED** [review]

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 2]
```diff
@@ -5921,6 +5921,8 @@ pi_result _pi_queue::synchronize() {
     return PI_SUCCESS;
 
   // For in-order queue just wait for the last command.
+  // If event is discarded then it can be in reset state or underlying level
+  // zero handle can have device scope, so we can't synchronize the last event.
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:5925]:
```suggestion
  // zero handle may not have device scope, so we can't synchronize the last event.
```

Comment (againull) [reply sycl/plugins/level_zero/pi_level_zero.cpp:5925]: If it has device scope then we can't synchronize, right? And if it has host scope then we can synchronize.

Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:5925]: Right, I read "device scope" as "host scope" for some reason. Nevermind.

**Review by smaslov-intel — APPROVED** [review]

**Comment by againull** [comment]
Basic/memory-consumption.cpp fail is known: https://github.com/intel/llvm/issues/9008
Device not found errors on SYCL Pre Commit / Linux / ESIMD Emu LLVM Test Suite (pull_request_target) expected to be fixed by https://github.com/intel/llvm/commit/305da175b77b20174b713fcb16b2003d9aabf17a

**Pull request merged**